_posts/2020-12-01-Fortran-Newsletter-December-2020.md

```diff
@@ -0,0 +1,149 @@
+* [#221](https://github.com/fortran-lang/fpm/pull/221),
+  [#239](https://github.com/fortran-lang/fpm/pull/239): Runner options for test and app executables.
+* [#220](https://github.com/fortran-lang/fpm/pull/220): Implement compiler and flags settings in Haskell fpm.
+* [#209](https://github.com/fortran-lang/fpm/pull/209):
+  [#237](https://github.com/fortran-lang/fpm/pull/237): Developer API docs.
+* [#216](https://github.com/fortran-lang/fpm/pull/216),
+  [#225](https://github.com/fortran-lang/fpm/pull/225),
+  [#226](https://github.com/fortran-lang/fpm/pull/226),
+  [#229](https://github.com/fortran-lang/fpm/pull/229),
+  [#236](https://github.com/fortran-lang/fpm/pull/236),
+  [#240](https://github.com/fortran-lang/fpm/pull/240),
+  [#247](https://github.com/fortran-lang/fpm/pull/240): Other fixes and improvements.
+
+Work in progress:
+
+* [First beta release](https://github.com/fortran-lang/fpm/milestone/1) (WIP): First feature-complete release of the Fortran implementation.
+* (WIP) [#230](https://github.com/fortran-lang/fpm/pull/230),
+        [#261](https://github.com/fortran-lang/fpm/pull/261): Specification of the fpm CLI.
+* (WIP) [#232](https://github.com/fortran-lang/fpm/pull/232): Allowing the `extra` section in fpm.toml.
+* (WIP) [#248](https://github.com/fortran-lang/fpm/pull/248): Refactor backend for incremental rebuilds.
+* (WIP) [#251](https://github.com/fortran-lang/fpm/pull/251): Dependency management.
+* (WIP) [#255](https://github.com/fortran-lang/fpm/pull/255): Setting the compiler and specifying test or app target.
+* (WIP) [#257](https://github.com/fortran-lang/fpm/pull/257): Implement `fpm install`.
+* (WIP) [#260](https://github.com/fortran-lang/fpm/pull/260): Fix CI to test release build.
+
+fpm is still in early development and we need as much help as we can get.
+Here's how you can help today:
+
+* Use it and let us know what you think! Read the [fpm packaging guide](https://github.com/fortran-lang/fpm/blob/master/PACKAGING.md) to learn how to build your package with fpm, and the [manifest reference](https://github.com/fortran-lang/fpm/blob/master/manifest-reference.md) to learn what are all the things that you can specify in the fpm.toml file.
+* Browse the [open issues](https://github.com/fortran-lang/fpm/issues) and see if you can help implement any fixes or features.
+* Adapt your Fortran package for fpm and submit it to the [Registry](https://github.com/fortran-lang/fpm-registry).
+* Improve the documentation.
+
+The short-term goal of fpm is to make development and installation of Fortran packages with dependencies easier.
+Its long term goal is to build a rich and decentralized ecosystem of Fortran packages and create a healthy
+environment in which new open source Fortran projects are created and published with ease.
+
+## Compilers
+
+### Classic Flang
+
+TODO @gklimowicz
+
+### LLVM Flang
+
+TODO @gklimowicz
+
```

Comment:
Work continues on LLVM Flang, concentrating on semantics, lowering and runtime. Significant contributions are being made for OpenMP and OpenACC support.

In conjunction with the MLIR-based code from the _fir-dev_ fork (the Fortran IR used for lowering), Flang can compile and run most F77 programs, including the Fortran Compiler Validation Suite (FCVS).

Pat McCormick is (still) working on an RFC for the merge of the lowering code
in the fir-dev fork into LLVM master. (This was interrupted by Supercomputing 2020 and other ECP duties.) The goal is to expedite this in a way that is acceptable to the Flang community, so we can do further work in the single master branch.

Recent updates include:
* Johannes Doerfert has created a web page at https://flang.llvm.org; you can find call and Slack logistics there
* Nichols Romero has an llvm-dev RFC for adding Fortran tests to the llvm-tests project: http://lists.llvm.org/pipermail/llvm-dev/2020-November/146873.html
* Andzrej Warzynski has a flang-dev RFC regarding flang option names: http://lists.llvm.org/pipermail/flang-dev/2020-November/000588.html
* Andzrej Warzynski has a cfe-dev RFC regarding refactoring clang to help flang driver become independent of clang: http://lists.llvm.org/pipermail/cfe-dev/2020-November/067263.html
* Changed representation of CHARACTER data in type system to make more consistent with other types (for arrays)
* Changed COMPLEX expression representation to provide better handling in lowering
* More improvements for supporting Fortran 77 programs
* Implemented runtime support for basic ALLOCATE/DEALLOCATE and further work
* Continued implementation of table-driven runtime for derived types; posted documentation
* Continued implementation of array expression lowering
* Improved error checks on forward references
* More updates to flang driver (option handling; -E can now be used to invoke just the Flang preprocessor)
* OpenACC semantic checks for modifiers on enter/exit data, set directives
* OpenACC lowering (enter/exit data, update, init, shutdown, wait directives)
* OpenMP structure checker updates; semantic checks for copyin clause; schedule class